source/Engine.cpp

```diff
@@ -504,6 +504,12 @@ void Engine::Step(bool isActive)
 	wasActive = isActive;
 	Audio::Update(center);
 
+	// Now that the calculation thread it is safe to update to the new zoom value.
+	if(nextZoom)
+	{
+		zoom = nextZoom;
+		nextZoom = 0.;
+	}
```

Comment:
Seems like maybe we should:

```suggestion
	// TODO: std::exchange
	swap(zoom, nextZoom);
	nextZoom = 0.; 
```
and go on with it.

Reply:
Just realized that this doesn't actually work, we still need the if or else `zoom` will always get overwritten with .0.

Reply:
Yeah I was thinking that nextZoom was getting set every frame. Oops!